### PR TITLE
 refactor(kv): delete deprecated kv service code

### LINKED ISSUE #18498: Remove references to tenant services from kv package
This is a list of the work necessary to remove the org, urm, user, password, and bucket code from the kv package now that these services have been re-implemented in the tenant service. I have included files and code links where kv code references these services, as well as any relevant details for taking this code out. This code can be broken out into separate issues and worked on simultaneously.

- [x] dashboards
- [x] auth
     - this code has already been moved to its own package, so this one will be solved once the feature flag is removed for that package
- [x] checks - in progress
    - `FindOrganizationByName` is used in Find and Create functions
     - URMs are used in `FindChecks`
- [x] lookup service 
   - This service references buckets, orgs, users, dashboards, and telegrafs. 
- [x] tasks 
     - requires a migration to put owner IDs on tasks that do not have them: #18503
- [ ] document & document options 
   - Replaced by pkger, will be removed after https://github.com/influxdata/influxdb/issues/18245
- [ ] remove org, user, urm, password, bucket functions from kv service definition

Many of the references to the org service in kv are via a function `FindOrganizationByName` which is in kv/org.go but is not found in the definition of an `influxdb.OrganizationService` nor the tenant service implementation of it. This function may need to be added to the tenant service.

### PR DESCRIPTION
**Pull request by GeorgeMac**
Closes #18498

This is a replacement of the initial spike: https://github.com/influxdata/influxdb/pull/19783

This includes removal of **a lot** of `*kv.Service` and `*http.Service` responsibilities:

- documents (update: I had to re-instate parts of this because as it turns out; this is still used by the UI. I was under the impression it was no longer dependended upon)
- telegrafs (will be delivered in PR above)
- notification rules + endpoints (delivered already in two separate PRs)
- checks
- orgs
- users
- buckets
- passwords
- URMs
- labels
- authorizations
- dashboards (will be delivered in PR above)
- secrets

It moves a lot of responsibility solely onto the `tenant` package as intended.
Along the way I found some fixes to long-standing bugs in tests (e.g. label service clients weren't working). And made sure that everything that relies on tenant data, gets it from the `tenant` family of definitions. _A lot_ of things still relied on `kv.Service` implementations. Which should no longer be the case.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Review by brettbuddin — APPROVED** [review]

**Review by yquansah — APPROVED** [review]

**Comment by stuartcarnie** [comment]
@GeorgeMac I noticed a couple of e2e tests failed – I have restarted them, but this might be something that needs review.

File: cmd/influxd/launcher/launcher.go [hunk 1]
```diff
@@ -735,28 +735,23 @@ func (m *Launcher) run(ctx context.Context) (err error) {
...
-	}
-
-	flushers := flushers{}
+	var flushers flushers
 	switch m.storeType {
 	case BoltStore:
 		store := bolt.NewKVStore(m.log.With(zap.String("service", "kvstore-bolt")), m.boltPath)
 		store.WithDB(m.boltClient.DB())
 		m.kvStore = store
-		m.kvService = kv.NewService(m.log.With(zap.String("store", "kv")), store, serviceConfig)
```
Comment (stuartcarnie) [inline cmd/influxd/launcher/launcher.go:749]: 👏

File: cmd/influxd/launcher/launcher.go [hunk 2]
```diff
@@ -735,28 +735,23 @@ func (m *Launcher) run(ctx context.Context) (err error) {
...
 		if m.testing {
 			flushers = append(flushers, store)
 		}
+
 	case MemoryStore:
 		store := inmem.NewKVStore()
 		m.kvStore = store
-		m.kvService = kv.NewService(m.log.With(zap.String("store", "kv")), store, serviceConfig)
```
Comment (stuartcarnie) [inline cmd/influxd/launcher/launcher.go:756]: 👏

**Review by stuartcarnie — APPROVED** [review]
Stellar work, @GeorgeMac 🥇 🎖️ 🏅 

I did not review the 116 changed files. 

I decided to check out the branch, ran all the tests locally. Following that, I built influxd and influx and ran the binaries. I verified it worked with my existing setup and then went through a relatively broad number of manual tests to verify behaviour. It is by no means exhaustive, but I was very happy that I ran into zero problems.

These are the activities I performed:

- ✅ Ran various queries over an existing instance, wrote data and queried it
-  :white_check_mark: Setup a brand new instance
-  :white_check_mark: Created a new bucket, `new-bucket`
-  :white_check_mark: Created a scraper to `influxd` to scrape data to `new-bucket`
-  :white_check_mark: Ran Flux queries against this data
-  :white_check_mark: Created a dashboard to show this data
-  :white_check_mark: Used `influx` CLI tool to list buckets, authorisations
-  :white_check_mark: Setup local Telegraf config and wrote to `my-bucket`
-  :white_check_mark: Verified writes to `my-bucket` from running Telegraf
-  :white_check_mark: Setup DB/RP mapping to `my-bucket`
-  :white_check_mark: Configured v1 authorisation `v1user` and password with read and write access to `my-bucket`
-  :white_check_mark: Started Grafana and connect using `v1user` credentials
-  :white_check_mark: Successfully executed InfluxQL queries
-  :white_check_mark: Created and assigned a label, `testing` to a bucket and a dashboard

**Comment by bednar** [comment]
Hi All,

there are some glitch between current implementation of document store and swagger definition. 

Could you look at #20163?

**Pull request merged**